core/block_svg.ts

```diff
@@ -892,8 +867,7 @@ export class BlockSvg
     this.disposing = true;
     super.disposeInternal();
 
-    if (common.getSelected() === this) {
-      this.unselect();
+    if (getFocusManager().getFocusedNode() === this) {
```

Comment:
It looks like you’re missing the equivalent “defocus” call, is that intentional?